litgpt/api.py

```diff
@@ -0,0 +1,219 @@
+                - bnb.nf4, bnb.nf4-dq, bnb.fp4, bnb.fp4-dq: 4-bit quantization from bitsandbytes
+                - bnb.int8: 8-bit quantization from bitsandbytes
+                for more details, see https://github.com/Lightning-AI/litgpt/blob/main/tutorials/quantize.md
+            precision: Indicates the Fabric precision setting to use.
+                For instance, "32-true", "16-mixed", "16-true", "bf16-mixed", "bf16-true".
+                For more details, see https://lightning.ai/docs/fabric/stable/api/fabric_args.html#precision
+        """
+
+        if device_type not in {"cpu", "cuda", "auto"}:
+            raise ValueError(f"Invalid device_type: {device_type}. Must be one of 'cpu', 'cuda', or 'auto'.")
+
+        if device_type == "auto":
+            device_type = "cuda" if torch.cuda.is_available() else "cpu"
+
+        num_devices = calculate_number_of_devices(devices)
+
+        if num_devices > 1:
+            raise NotImplementedError(
+                "Support for multiple devices is currently not implemented, yet."
+            )
+
+        # It's called `model` and not `checkpoint_dir` in the function signature
+        # because we will later add functionality to automatically download the model
+        # E.g.,
+        #   model = "EleutherAI/pythia-16m", source = "hf"
+        #   will download the model from the HF hub if it doesn't exist locally under
+        #   "EleutherAI/pythia-16m" or "checkpoints/EleutherAI/pythia-16m"
+        # And
+        #   source = "EleutherAI/pythia-16m", hub = "local" will always consider the local model
+        # Also, we may add support for other hubs in the future.
+        checkpoint_dir = extend_checkpoint_dir(Path(model))
+
+        config = Config.from_file(checkpoint_dir / "model_config.yaml")
+        torch.set_float32_matmul_precision("high")
+
+        devices = _normalize_parse_gpu_input_to_list(devices, include_cuda=True, include_mps=True)
+        precision = precision or get_default_supported_precision(training=False)
+
+        fabric = L.Fabric(
+            accelerator=device_type,
+            devices=devices,
+            precision=precision,
+        )
+
+        checkpoint_path = checkpoint_dir / "lit_model.pth"
+        tokenizer = Tokenizer(checkpoint_dir)
+
+        prompt_style = (
+            load_prompt_style(checkpoint_dir)
+            if has_prompt_style(checkpoint_dir)
+            else PromptStyle.from_config(config)
+        )
+
+        with fabric.init_module(empty_init=(num_devices > 1)):
+            model = GPT(config)
+
+        with fabric.init_tensor():
+            model.set_kv_cache(batch_size=1)
+
+        model.eval()
+        model = fabric.setup_module(model)
+        load_checkpoint(fabric, model, checkpoint_path)
+        return cls(
+            model=model, tokenizer=tokenizer, devices=devices,
+            prompt_style=prompt_style, checkpoint_dir=checkpoint_dir, fabric=fabric,
+        )
+
+    def generate(
```

Comment:
can we also add `[include_prompt](https://github.com/Lightning-AI/litgpt/blob/b663a8185c8da81a768bcb59ed5996dcadccfb59/litgpt/generate/base.py#L91C5-L91C19)` argument here?

Reply:
Good catch, thanks! Just added it.